targets/mbed/source/jerry_run.cpp

```diff
@@ -17,107 +17,100 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-#include "jerry-core/jerry.h"
+#include "jerry-core/jerry-api.h"
 #include "jerry_extapi.h"
 #include "jerry_run.h"
 
 static const char* fn_sys_loop_name = "sysloop";
 
 int js_entry (const char *source_p, const size_t source_size)
 {
-  const jerry_api_char_t *jerry_src = (const jerry_api_char_t *) source_p;
-  jerry_completion_code_t ret_code = JERRY_COMPLETION_CODE_OK;
-  jerry_flag_t flags = JERRY_FLAG_EMPTY;
-  jerry_api_object_t *err_obj_p = NULL;
-  jerry_api_value_t err_value = jerry_api_create_void_value ();
-
-  jerry_init (flags);
+  const jerry_char_t *jerry_src = (const jerry_char_t *) source_p;
+  jerry_init (JERRY_INIT_EMPTY);
+  uint8_t ret_code = 0;
+  jerry_value_t ret_value = jerry_create_undefined ();
 
   js_register_functions ();
 
-  if (!jerry_parse (jerry_src, source_size, &err_obj_p))
+  jerry_value_t parsed_code = jerry_parse (jerry_src, source_size, false);
+
+  if (!jerry_value_has_error_flag (parsed_code))
   {
-    printf ("Error: jerry_parse failed\r\n");
-    ret_code = JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION;
-    jerry_api_release_object (err_obj_p);
+    ret_value = jerry_run (parsed_code);
+
+    jerry_release_value (ret_value);
   }
   else
   {
-    if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
-    {
-      ret_code = jerry_run (&err_value);
-      jerry_api_string_t *err_str_p = NULL;
-
-      if (err_str_p != NULL)
-      {
-        jerry_api_release_string (err_str_p);
-      }
-    }
+    printf("Error: jerry_parse failed!\r\n");
+    ret_code = JERRY_ERROR_SYNTAX;
```

Comment:
This is misleading and cannot be guaranteed that  JERRY_ERROR_SYNTAX won't be 0.